reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md

```diff
@@ -0,0 +1,777 @@
+/// </summary>
+/// <param name="context">The context for the feedback call.</param>
+/// <param name="token">The cancellation token to cancel the operation.</param>
+/// <returns>The feedback item.</returns>
+public FeedbackItem? GetFeedback(FeedbackContext context, CancellationToken token)
+{
+    // Target describes the different kinds of triggers to activate on,
+    var target = context.Trigger;
+    var commandLine = context.CommandLine;
+    var ast = context.CommandLineAst;
+
+    // defining the header and footer variables
+    string header;
+    string footer;
+
+    // List of the actions
+    List<string>? actions = new List<string>();
+
+    // Trigger on success code goes here
+
+    // Trigger on error code goes here
+
+    return null;
+}
+```
+
+The following image shows how these fields are used in the suggestions that are displayed to the
+user.
+
+![Screenshot of example feedback providers][06]
+
+### Create suggestions for a Success trigger
+
+For a successful invocation, we want to expand any aliases used in the last execution. Using the
+`CommandLineAst`, we identify any aliased commands and create a suggestion to use the fully
+qualified command name instead.
+
+```csharp
+// Trigger on success
+if (target == FeedbackTrigger.Success){
+    // Getting the commands from the AST and only finding those that are Commands
+    var astCmds = ast.FindAll((cAst) => cAst is CommandAst, true);
+
+    // Inspect each of the commands
+    foreach(var command in astCmds)
+    {
+
+        // Get the command name
+        var aliasedCmd = ((CommandAst) command).GetCommandName();
+
+        // Check if its an alias or not, if so then add it to the list of actions
+        if(TryGetAlias(aliasedCmd, out string commandString))
+        {
+            actions.Add(aliasedCmd + " --> " + commandString);
+        }
+
+    }
+
+    // If no alias was found return null
+    if(actions.Count == 0)
+    {
+        return null;
+    }
+
+    // If aliases are found, set the header to a description and return a new FeedbackItem.
+    header = "You have used an aliased command:";
+    // Copy actions to _candidates for the predictor
+    _candidates = actions;
+
+    return new FeedbackItem(header, actions);
+}
+```
+
+### Implement the TryGetAlias() method
+
+The `TryGetAlias()` method is a private helper function that returns a boolean value to indicate
+whether the command is an alias. In the class constructor, we created a PowerShell instance that we
+can use to run PowerShell commands. The `TryGetAlias()` method uses this PowerShell instance to
+invoke the `GetCommand` method to determine if the command is an alias. The `AliasInfo` object
+returned by `GetCommand` contains full name of the aliased command. The parameters of
+`TryGetAlias()` are: `command`, which is the command to be checked, and `targetCommand`, which is
+the output variable to contain the full name of the aliased command.
```

Comment:
```suggestion
returned by `GetCommand` contains full name of the aliased command.
```
The last sentence can be removed given that the xml doc comment of the method gives the same information.